core/unit_test/view/TestBasicView.hpp

```diff
@@ -0,0 +1,249 @@
+
+  view_type view("test_view", extents);
+
+  EXPECT_TRUE(view.data_handle().has_record());
+  EXPECT_NE(view.data_handle().get(), nullptr);
+  EXPECT_EQ(view.extents(), extents);
+  EXPECT_EQ(view.data_handle().use_count(), 1);
+  EXPECT_TRUE(view.is_exhaustive());
+  EXPECT_EQ(view.data_handle().get_label(), "test_view");
+  size_t expected_size = 1;
+  // Avoid pointless comparison of unsigned warning for rank==0
+  for (int r = 0; r < static_cast<int>(view_type::rank()); r++)
+    expected_size *= extents.extent(r);
+  EXPECT_EQ(view.size(), expected_size);
+  EXPECT_EQ(view.empty(), expected_size == 0u);
+}
+
+template <class T, template <std::size_t> class LayoutType, class ExtentsType>
+static void test_mapping_constructor(const ExtentsType &extents,
+                                     std::size_t padding) {
+  using extents_type  = ExtentsType;
+  using layout_type   = LayoutType<Kokkos::dynamic_extent>;
+  using mapping_type  = typename layout_type::template mapping<ExtentsType>;
+  using accessor_type = Kokkos::Impl::CheckedReferenceCountedAccessor<
+      T, typename ExecutionSpace::memory_space>;
+  using view_type =
+      Kokkos::BasicView<T, extents_type, layout_type, accessor_type>;
+  static_assert(std::is_same_v<typename view_type::mapping_type, mapping_type>);
+
+  auto mapping = mapping_type(extents, padding);
+
+  view_type view("test_view", mapping);
+
+  EXPECT_TRUE(view.data_handle().has_record());
+  EXPECT_NE(view.data_handle().get(), nullptr);
+  EXPECT_EQ(view.data_handle().use_count(), 1);
+  EXPECT_EQ(view.data_handle().get_label(), "test_view");
+  EXPECT_EQ(view.extents(), mapping.extents());
+  EXPECT_EQ(view.is_exhaustive(), mapping.is_exhaustive());
+  size_t expected_size = 1;
+  // Avoid pointless comparison of unsigned warning for rank==0
+  for (int r = 0; r < static_cast<int>(view_type::rank()); r++)
+    expected_size *= view.extent(r);
+  EXPECT_EQ(view.size(), expected_size);
+  EXPECT_EQ(view.empty(), expected_size == 0u);
+}
+
+template <class ViewType>
+struct MDRangeTestFunctor {
+  ViewType view;
+  template <class... Idxs>
+  KOKKOS_FUNCTION void operator()(Idxs... idxs) const {
+    view(idxs...) = (idxs + ...);
+  }
+};
+
+template <class T, class LayoutType, class ExtentsType>
+static void test_access_with_extents(const ExtentsType &extents) {
+  using extents_type  = ExtentsType;
+  using layout_type   = Kokkos::Experimental::layout_right_padded<>;
+  using accessor_type = Kokkos::Impl::CheckedReferenceCountedAccessor<
+      T, typename ExecutionSpace::memory_space>;
+  using view_type =
+      Kokkos::BasicView<T, extents_type, layout_type, accessor_type>;
+
+  auto view = view_type("test_view", extents);
+
+  EXPECT_TRUE(view.data_handle().has_record());
+  EXPECT_NE(view.data_handle().get(), nullptr);
+
+  auto mdrange_policy =
+      make_spanning_mdrange_policy_from_extents<ExecutionSpace>(extents);
+
+  Kokkos::parallel_for(mdrange_policy, MDRangeTestFunctor<view_type>{view});
```

Comment:
You don't want to assert anything?
What does "access with extents" mean?

Reply:
its "test_access" function "with extents" added.